src/terminal/char_grid/cursor.rs

```diff
@@ -24,10 +24,20 @@ pub struct Cursor {
     pub coords: Coords,
     pub style: Styles,
     pub text_style: Styles,
+    config: Config,
```

Comment:
> I was thinking that rather than copying the Config around everywhere or having it owned by Terminal and referenced elsewhere as &Config that we could do something like allocate new Configs on the heap using either Rc<Config> or Arc<Config>--if I understand correctly, the choice depends on whether we need the Config to be available in more than one context. (EDIT: i mean more than one threaded context)

I think it would make sense to put the Config in an Rc, and have the `Terminal` and each `CharGrid` have a separate instance of it, but to pass it by reference to each function that needs it on any of the components of `CharGrid`. Reference counting does have a cost, we don't want to have to pay it every time we create a new character cell, but creating new grids is a much less frequent thing.

I think `Rc` is fine instead of `Arc` here.

> what do you think about storing the Config on the CharGrid instead and relying on the Terminal to deref to CharGrid to access it? Does that even make sense? 

It's what I thought we could do, but it doesn't work.

Right now, the Terminal dereferences to the active CharGrid. So you can call `terminal.move_cursor(movement)` and the cursor in the active char grid will get moved. I thought we could have the Terminal dereference to some temporary struct that held references to the CharGrid and the Config, but that doesn't work because you can't dereference to something that didn't already exist.